### PR TITLE
CMake FindMallocMC: Honor Paths, Outdated Versions

### PR DESCRIPTION
**Pull request by ax3l**
If an outdated version of mallocMC was found, the second search was not successful. Also, CMake 2.8.12.2+ is required (spotted by @PrometheusPi, thx!).
  ComputationalRadiationPhysics/cmake-modules#8

Some temporary variables have been cleaned up.
  ComputationalRadiationPhysics/cmake-modules#9

This pull request increases the CMake dependencies of PIConGPU to 2.8.12.2+ (the last 2.X release of CMake, using already a >=3 version will also work fine). Docs and `CMakeLists.txt` are updated with this pull.

--

Third party update for `FindmallocMC.cmake` in `thirdParty/cmake-modules` performed via

```bash
GIT_AUTHOR_NAME="Third Party" GIT_AUTHOR_EMAIL="picongpu@hzdr.de" \
    git subtree pull --prefix thirdParty/cmake-modules/ \
    git@github.com:ComputationalRadiationPhysics/cmake-modules.git dev --squash
```

### CODE REVIEW

**Comment by ax3l** [comment]
@BenjaminW3 do you like to pick up this PR? :)

File: thirdParty/cmake-modules/FindmallocMC.cmake [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # To provide a hint to this module where to find the mallocMC installation,
 # set the MALLOCMC_ROOT environment variable. You can also set the
 # MALLOCMC_ROOT CMake variable, which will take precedence over the environment
-# variable.
+# variable. Boths hints are preferred over the DEFAULT_PATHS.
```
Comment (BenjaminW3) [inline thirdParty/cmake-modules/FindmallocMC.cmake]: Both~~s~~

Comment (ax3l) [inline thirdParty/cmake-modules/FindmallocMC.cmake]: oh no, that's in the subtree. I won't fix it in this pull but will add a PR upstream.

File: thirdParty/cmake-modules/FindmallocMC.cmake [hunk 2]
```diff
@@ -22,6 +22,8 @@
 #   mallocMC_INCLUDE_DIRS    - Include directories for the mallocMC headers.
 #   mallocMC_FOUND           - TRUE if FindmallocMC found a working install
 #   mallocMC_VERSION         - Version in format Major.Minor.Patch
+# and for individual COMPONENTS:
+#   mallocMC_<COMPONENT>_FOUND  - TRUE if FindmallocMC found that component
```
Comment (BenjaminW3) [inline thirdParty/cmake-modules/FindmallocMC.cmake]: Could we add a note which case the resulting ```<COMPONENT>``` part of the string will have? Is it original case or uppercase or will both be defined?

Comment (ax3l) [inline thirdParty/cmake-modules/FindmallocMC.cmake]: original case as the [recommendation](http://www.cmake.org/cmake/help/v3.0/module/FindPackageHandleStandardArgs.html), good point. will also add this upstream (see link above).

**Pull request merged**